### PR TITLE
Check for organization Id in request attributes

### PR DESCRIPTION
**Pull request by shanggeeth**
### Changes
- Add approles as a default claim
- Check for organization Id in request attributes

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
     protected String getScope(String scope, Map<String, String> authenticatorProperties) {
 
         if (StringUtils.isBlank(scope)) {
-            scope = "openid email profile phone address";
+            scope = "openid email profile phone address approles";
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java:169]: `approles` should be added to 171 as well to do RBAC

Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java]: Addressed in 795552bf02f3700a4b382965ed59673dc83411a3
Changed to correct name app_roles

File: components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java [hunk 2]
```diff
@@ -166,8 +167,9 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
     protected String getScope(String scope, Map<String, String> authenticatorProperties) {
 
         if (StringUtils.isBlank(scope)) {
-            scope = "openid email profile phone address";
+            scope = "openid email profile phone address app_roles";
         }
+        scope = addAppRolesScope(scope);
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java:172]:
```suggestion
        else {
              scope = addAppRolesScope(scope);
        }
```

File: components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java [hunk 3]
```diff
@@ -260,7 +262,12 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
         }
 
         // First priority for organization Id.
-        if (request.getParameterMap().containsKey(ORG_ID_PARAMETER)) {
+        if (StringUtils.isNotBlank((String) request.getAttribute(ORG_ID_PARAMETER))) {
+            String organizationId = (String) request.getAttribute(ORG_ID_PARAMETER);
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java:266]: Shall we add a comment here?

**Review by ShanChathusanda93 — DISMISSED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java (hunk 2 above)
Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java]: Addressed in a94508f0e7c82fb09c4db34e981c2a445742c047

File: components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java (hunk 3 above)
Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java:271]: Addressed in a94508f0e7c82fb09c4db34e981c2a445742c047

File: components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java [hunk 4]
```diff
@@ -166,7 +167,9 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
     protected String getScope(String scope, Map<String, String> authenticatorProperties) {
 
         if (StringUtils.isBlank(scope)) {
-            scope = "openid email profile phone address";
+            scope = "openid email profile phone address app_roles";
+        } else {
+            scope = addAppRolesScope(scope);
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java:172]: remove all these changes and return `addAppRolesScope(scope)`

Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java]: Addressed in 5593c7a9dac8801426e65c4c29b375c98c25c1db

**Review by sadilchamishka — APPROVED** [review]

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**